_posts/2015-12-16-29.markdown

```diff
@@ -0,0 +1,283 @@
+
+If you don’t understand computers, you might want to skip this section. A TLDR is at the bottom.
+
+This time last year, the GitHub Mac team released Carthage, it’s a de-centralised dependency manager for Cocoa projects. It throws away many ideas that I’ve worked towards for years:
+
+* Human-readable formats for understanding how a library works, sitting atop open platforms.
+* Discoverability of awesome libraries from people you’ve not heard of.
+* Working with existing codebases.
+* Actively engaging with the whole community for large infrastructure improvements.
+
+They choose to split the Cocoa community, forcing developers to make an “us vs them” choice. A chunk of community agrees with them and uses it. Now, in order to be discovered new contributors support CocoaPods, and in order to support as many users as they can they need to support Carthage. Raising the barrier to entry to OSS.
+
+I had hoped that the Cocoa community would have done what the Ruby community did a few years ago, and agree to not use GitHub’s replacement for the Rubygems infrastructure.
+
+It’s been personally disappointing, and something that I’ve avoided talking about publicly. I strive to public have an attitude of _be nice or shut up_, but that’s not very good for looking back in a few years trying to understand how I became who I am.
+
+I wasn’t happy to write about this, so I held off on starting  for a few weeks. I’m glad I did, something I had been waiting for years happened. When Apple open-source’d Swift, they announced a dependency manager, and have been doing a really job at taking the spirit of CocoaPods.
+
+> CocoaPods aims to improve the engagement with, and discoverability of, third party open-source Cocoa libraries.
+
+They’re actively talking with the community and us, finding ways to share work and to promote collaboration. It’s eased a lot of the friction I had been feeling. Apple has a fresh start in a way we could not, and will be able to do things much better than the CocoaPods team could.
+
+All in all, when people have been asking what I want to do in the next year, I’m feel pretty done on the “lowering the barrier to entry” space, this year it became less fun.
+
+*TLDR* - People wrote a competitor for something I work on, it has different ideas that aren’t about community spirit, people use it and it makes it harder for new people to contribute to the community I’m in, this bums me out. Ironically, Apple is potentially saving the day by doing their own competitor too. They’re not known for being good in this space among developers.
+
+With the awkward bit over, perhaps there’s space for some positivity now.
```

Comment:
I think it's so valuable to be honest. Thanks for sharing this. Sometimes it's hard when you know people so publicly and they always seem to be 100% happy and kicking ass. Amazing to know that you are human too and things get to you too. However, how you respond to that is just trying to move forward and be positive, rather than negatively complaining. If there's one thing people take away from this writeup, I hope it's that. <3